### PR TITLE
Controllers should always go async on start

### LINKED ISSUE #6272: Controllers process only sending sdnotify after obtaining a lease
Service definitions with `Type=notify` start successfully on the master which has obtained the lease and will fail (timeout) on all other masters.

### PR DESCRIPTION
**Pull request by smarterclayton**
Fixes #6272 @sdodson @abutcher

### CODE REVIEW

**Comment by smarterclayton** [comment]
[test]

**Comment by deads2k** [comment]
@smarterclayton we really need to be able to inspect status for individual controllers.

File: pkg/cmd/server/start/start_master.go [hunk 1]
```diff
@@ -395,18 +396,8 @@ func (m *Master) Start() error {
...
-					glog.Fatal(err)
-				}
-			}()
-		}
-		return nil
-	}
 
-	if m.controllers {
+	case m.controllers:
 		glog.Infof("Starting controllers on %s (%s)", m.config.ServingInfo.BindAddress, version.Get().String())
```
Comment (deads2k) [inline pkg/cmd/server/start/start_master.go]: Why shouldn't I have this message even when the controllers are starting with the API server?

Comment (smarterclayton) [inline pkg/cmd/server/start/start_master.go]: The controllers print their own messages - this is the summarization message about what mode are you in.

**Comment by smarterclayton** [comment]
Define status?

**Comment by deads2k** [comment]
> Define status?

Which controllers are functioning.  Its currently impossible to check if a particular controller is alive, which becomes annoying when you're dealing with, "why isn't my RC producing pods?"

File: pkg/cmd/server/start/start_master.go (hunk 1 above)
Comment (deads2k) [inline pkg/cmd/server/start/start_master.go]: > The controllers print their own messages - this is the summarization message about what mode are you in.

ok

**Comment by deads2k** [comment]
lgtm [merge]

**Comment by smarterclayton** [comment]
Hrm - part of this should be handled by GET /controllers

**Comment by smarterclayton** [comment]
[merge]

**Comment by smarterclayton** [comment]
[test]

**Comment by smarterclayton** [comment]
[merge]

**Pull request merged**